streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java

```diff
@@ -2330,4 +2348,266 @@ public void shouldNotAllowNullNamedOnProcessWithStores() {
+        table1.join(table2, MockValueJoiner.TOSTRING_JOINER).toStream().to(output);
+
+        final Collection<Set<String>> copartitionGroups =
+            TopologyWrapper.getInternalTopologyBuilder(builder.build()).copartitionGroups();
+
+        assertEquals(1, copartitionGroups.size());
+        assertEquals(new HashSet<>(Arrays.asList(input1, input2)), copartitionGroups.iterator().next());
+
+        try (final TopologyTestDriver driver = new TopologyTestDriver(builder.build(), props)) {
+            final TestInputTopic<String, String> inputTopic1 =
+                driver.createInputTopic(input1, Serdes.String().serializer(), Serdes.String().serializer(), Instant.ofEpochMilli(0L), Duration.ZERO);
+            final TestInputTopic<String, String> inputTopic2 =
+                driver.createInputTopic(input2, Serdes.String().serializer(), Serdes.String().serializer(), Instant.ofEpochMilli(0L), Duration.ZERO);
+            final TestOutputTopic<String, String> outputTopic =
+                driver.createOutputTopic(output, Serdes.String().deserializer(), Serdes.String().deserializer());
+
+            inputTopic2.pipeInput("1", "stream");
+
+            assertThat(
+                outputTopic.readKeyValuesToMap(),
+                is(emptyMap())
+            );
+
+            inputTopic1.pipeInput("1", "table");
+
+            final Map<String, String> expected = mkMap(
+                mkEntry("1", "table+stream")
+            );
+            assertThat(
+                outputTopic.readKeyValuesToMap(),
+                is(expected)
+            );
+        }
+    }
+
+    @Test
+    public void shouldKStreamToKTableLeftJoin() {
+        final Consumed<String, String> consumed = Consumed.with(Serdes.String(), Serdes.String());
+        final StreamsBuilder builder = new StreamsBuilder();
+
+        final String input1 = "input1";
+        final String input2 = "input2";
+        final String output = "output";
+
+        final KTable<String, String> leftTable = builder.stream(input1, consumed).toTable();
+        final KTable<String, String> rightTable = builder.table(input2, consumed);
+        final KTable<String, String> joined = leftTable.leftJoin(rightTable, MockValueJoiner.TOSTRING_JOINER);
+        joined.toStream().to(output);
+        checkLeftJoin(builder.build(), input1, input2, output);
+    }
+
+    @Test
+    public void shouldKStreamToKTableStreamLeftJoin() {
+        final String streamTopic = "streamTopic";
+        final String tableTopic = "tableTopic";
+        final String output = "output";
+        final StreamsBuilder builder = new StreamsBuilder();
+        final Consumed<String, String> consumed = Consumed.with(Serdes.String(), Serdes.String());
+
+        final KStream<String, String> stream = builder.stream(streamTopic, consumed);
+        final KTable<String, String> table =  builder.stream(tableTopic, consumed).toTable();
+
+        stream.leftJoin(table, MockValueJoiner.TOSTRING_JOINER).to(output);
```

Comment:
Why do you use a `leftJoin` and not an `join` (guess it does not make much of a difference -- fels just more natural to use inner join in all test)?